### DISCUSSION TITLE
Can you migrate Minio data to a new Kubernetes cluster?

### DISCUSSION THREAD

**Discussion by collimarco** [minio/minio#15581] [Q&A] [answered]
I am considering Minio as an alternative to S3.

However I don't understand how you can migrate Minio data from a Kubernetes cluster to a new one.

Sometimes you are forced to replace the K8s cluster with a new one for major version upgrades and I wonder how you can move all the data...

What is the correct approach?

**Comment by klauspost** [comment, marked as answer]
Hopefully upgrading K8s should leave you with an upgrade path. While I can't make guarantees on third party software, I also don't see them breaking everything currently deployed.

For replacing hardware you can [attach new hardware as a new server pool](https://docs.min.io/minio/baremetal/installation/expand-minio-distributed.html) and [decommission](https://docs.min.io/minio/baremetal/installation/decommission-pool.html) the old hardware. 

All of this is a zero downtime operation, except for a few seconds of restarts.
